hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1411,6 +1448,97 @@ protected AccessTokenProvider getTokenProvider() {
     return tokenProvider;
   }
 
+  public AzureBlobFileSystem getMetricFilesystem() throws IOException {
+    if (metricFs == null) {
+      try {
+        Configuration metricConfig = abfsConfiguration.getRawConfiguration();
+        String metricAccountKey = metricConfig.get(FS_AZURE_METRIC_ACCOUNT_KEY);
+        final String abfsMetricUrl = metricConfig.get(FS_AZURE_METRIC_URI);
+        if (abfsMetricUrl == null) {
+          return null;
+        }
+        metricConfig.set(FS_AZURE_ACCOUNT_KEY_PROPERTY_NAME, metricAccountKey);
+        metricConfig.set(AZURE_CREATE_REMOTE_FILESYSTEM_DURING_INITIALIZATION, "false");
+        URI metricUri;
+        metricUri = new URI(FileSystemUriSchemes.ABFS_SCHEME, abfsMetricUrl, null, null, null);
+        metricFs = (AzureBlobFileSystem) FileSystem.newInstance(metricUri, metricConfig);
```

Comment:
This won't create a chain since the only place where the metric filesystem is initialized is inside the client method.

Reply:
So is the suggestion that we don't create a new filesystem instance, just use the URL for execution ?

Reply:
Yes, because we donot want any functionality of fileSystem, but just want to trigger a server call to metric uri and would remove circular dependency. In the customized method, it take metricUri, create the restOperation and then execute it.

Reply:
The primary intention behind having a distinct configuration for the metric file system is to offer users the flexibility to choose where the additional metric calls are directed. Users can configure the system to send metrics either to the existing file system or to a separate file system based on their preferences. If a separate account is utilized for the metric URI, authentication may fail because we are currently employing the existing AbfsClient instance.

Reply:
Distinct configuration is still required for the metricURI and the cred for the metric-storage FS, which is not negated in the comment.
Comment was about creating a full fileSystem object to just hit one uri and about the implementation bug.

+1 on metricUri and cred. Current impl at head commit https://github.com/apache/hadoop/pull/6314/commits/5696ba64f1d90eadc5e40d8e70808743a3174819 looks amazing!